lib/ansible/modules/lineinfile.py

```diff
@@ -267,7 +267,7 @@ def present(module, dest, regexp, line, insertafter, insertbefore, create,
             try:
                 os.makedirs(b_destpath)
             except Exception as e:
-                module.fail_json(msg='Error creating %s Error code: %s Error description: %s' % (b_destpath, e[0], e[1]))
+                module.fail_json(msg='Error creating %s Error description: %s' % (b_destpath, to_native(e)))
```

Comment:
This is a good fix. Amazing we haven't hit this sooner.

I'm wondering if we should also convert the `b_destpath` to text as well.

```suggestion
                module.fail_json(msg='Error creating %s Error description: %s' % (to_text(b_destpath), to_native(e)))
```

Reply:
good call, should probably be text on both- updated (and tweaked the error formatting a bit)